src/Orc.Snapshots/Services/SnapshotStorageServiceBase.cs

```diff
@@ -84,19 +84,19 @@ protected virtual async Task<byte[]> ConvertSnapshotToBytesAsync(ISnapshot snaps
         {
             using (var memoryStream = new MemoryStream())
             {
-                using (var zip = new ZipFile())
+                using (var archive = new ZipArchive(memoryStream, ZipArchiveMode.Create))
                 {
                     var metadata = new Dictionary<string, string>();
                     metadata["title"] = snapshot.Title;
                     metadata["category"] = snapshot.Category ?? string.Empty;
                     metadata["created"] = snapshot.Created.ToString("yyyy-MM-dd HH:mm:ss");
 
-                    zip.AddEntry("metadata", GetMetadataBytes(metadata));
+                    var metadataEntry = archive.CreateEntry("metadata");
+                    metadataEntry.OpenAndWrite(GetMetadataBytes(metadata));
 
                     var snapshotBytes = await snapshot.GetAllBytesAsync();
-                    zip.AddEntry("data", snapshotBytes);
-
-                    zip.Save(memoryStream);
+                    var dataEntry = archive.CreateEntry("data");
+                    dataEntry.OpenAndWrite(snapshotBytes);
                 }
 
                 return memoryStream.ToArray();
```

Comment:
Will not lead to exception because of ToArray() works with steam's buffer, but i assume memoryStream closed after going outside of ZipArchive using block (you're using overload which not keeps stream open after disposing ZipArchive here, which is different from other places in PR.